src/components/PeopleTable.tsx

```diff
@@ -1,4 +1,101 @@
-export const PeopleTable = () => {
+import classNames from 'classnames';
+import React from 'react';
+import {
+  Link, useLocation, useParams, useSearchParams,
+} from 'react-router-dom';
+import { Person } from '../types/Person';
+import { SearchLink } from './SearchLink';
+
+type Props = {
+  people: Person[];
+};
+
+export const PeopleTable:React.FC<Props> = ({ people }) => {
+  const { slug } = useParams();
+  const isSelected = (peop: Person) => peop.slug === slug;
+  const location = useLocation();
+
+  const [searchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || null;
+  const query = searchParams.get('query') || '';
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+  const centuries = searchParams.getAll('centuries');
+
+  let newPeople = [...people];
+
+  const filtersOnSex = () => {
+    if (sex) {
+      newPeople = newPeople.filter(a => a.sex === sex);
+    }
+  };
+
+  const filterOnQuery = () => {
+    newPeople = newPeople.filter(
+      a => a.name.toLocaleLowerCase().includes(query.toLocaleLowerCase())
+        || a.motherName?.toLocaleLowerCase().includes(
+          query.toLocaleLowerCase(),
+        )
+        || a.fatherName?.toLocaleLowerCase().includes(
+          query.toLocaleLowerCase(),
+        ),
+    );
```

Comment:
good, but difficult to read, divide the expressions into variables whose names will explain what is in them